src/main/java/blf/core/writers/XesWriter.java

```diff
@@ -174,38 +161,32 @@ public void addStringList(@NonNull String key, @NonNull List<String> values) {
     }
 
     @Override
-    protected void writeState(String filenameSuffix) throws Throwable {
+    protected void writeState(String fileNameSuffix) {
         LOGGER.info("Xes export started.");
-        // Before
-        Map<String, Map<String, XTrace>> formerTraces = deepCopyTraces();
-        Map<String, Map<String, Map<String, XEvent>>> formerEvents = deepCopyEvents();
-        // Modification of state
         final Map<String, XLog> logs = this.getLogs();
-        // Restore of state
-        this.traces.clear();
-        this.events.clear();
-        this.traces.putAll(formerTraces);
-        this.events.putAll(formerEvents);
```

Comment:
Without this storing and retrieving of the former state, the safe batching mode will fail. This is because getLogs() modifies XesWriters state.
In both streaming and safe batching mode, this function writeState() gets called multiple times. But getLogs() modifies the state in a way which assumes that it is only called once at the very end of the extraction.
It may not be pretty, but without it, safe batching at least will not produce the same result as default batching!